### PR TITLE
Initial stab at StaffBot.

### PR DESCRIPTION
**Pull request by kkamalov**

### CODE REVIEW

File: orchestra/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # The current Orchestra version.
-__version__ = '0.1.42'
+__version__ = '0.1.43'
```
Comment (joshblum) [inline orchestra/__init__.py]: why?

File: orchestra/core/errors.py [hunk 2]
```diff
@@ -60,3 +60,7 @@ class WorkflowError(Exception):
 
 class TimerError(Exception):
     pass
+
```
Comment (joshblum) [inline orchestra/core/errors.py]: make `bots/errors.py` ?

File: orchestra/management/commands/runstaffbot.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    def handle(self, *args, **options):
+        autoreload.main(self.inner_run, None, options)
+
+    def inner_run(self, *args, **options):
+        slack_client = SlackClient(settings.STAFF_BOT_API_KEY)
+
+        if slack_client.rtm_connect():
+            staff_bot = StaffBot(slack_client)
+            while True:
```
Comment (joshblum) [inline orchestra/management/commands/runstaffbot.py:23]: how would we use this in production? what port is it listening on? do we have to spin up another server?

File: example_project/example_project/settings.py [hunk 4]
```diff
@@ -112,4 +112,8 @@
 STATIC_URL = '/static/'
 STATIC_ROOT = 'staticfiles'
 
+
+# Slack StaffBot API token
```
Comment (marcua) [inline example_project/example_project/settings.py]: staffbot throughout

File: orchestra/management/commands/runstaffbot.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class Command(BaseCommand):
+    help = 'Runs a Slack StaffBot for RTM calls'
+
+    def handle(self, *args, **options):
+        autoreload.main(self.inner_run, None, options)
+
+    def inner_run(self, *args, **options):
+        slack_client = SlackClient(settings.STAFF_BOT_API_KEY)
```
Comment (marcua) [inline orchestra/management/commands/runstaffbot.py]: We currently use `slacker` for other Slack API-related work.  Does the RTM class there do what we want?

https://github.com/os/slacker/blob/256b21bc45fda9584a30158360ed9d735b20c53e/slacker/__init__.py#L482

File: orchestra/__init__.py (hunk 1 above)
Comment (kkamalov) [inline orchestra/__init__.py]: removed

File: orchestra/core/errors.py (hunk 2 above)
Comment (kkamalov) [inline orchestra/core/errors.py]: done

File: orchestra/management/commands/runstaffbot.py (hunk 3 above)
Comment (kkamalov) [inline orchestra/management/commands/runstaffbot.py:23]: discussed offline

File: orchestra/management/commands/runstaffbot.py (hunk 5 above)
Comment (kkamalov) [inline orchestra/management/commands/runstaffbot.py]: unfortunately slacker on it's own does not support RTM: https://github.com/os/slacker/issues/17

**Comment by joshblum** [comment]
lgtm after rebase

**Pull request merged**